plugin/ViewFieldModifier/EstateViewFieldModifierTypeTitle.php

```diff
@@ -65,6 +65,27 @@ public function getAPIFields(): array
 		return array_values(array_unique(array_merge($this->_viewFields, $titleFields)));
 	}
 
+	/**
+	 *
+	 * @return array
+	 *
+	 */
+
+	public function getAPICustomFields(): array
+	{
+		$titleFields = [
+			'objekttitel',
+			'objektbeschreibung',
+			'ort',
+			'plz',
+			'objektart',
+			'vermarktungsart',
+			'Id'
+		];
+
+		return array_values(array_unique(array_merge($this->_viewFields, $titleFields)));
```

Comment:
Since the custom fields can only use the fields from `$titleFields`, why are we merging them with other fields?

Reply:
Because if it's not in the $this->_viewFields section it will not display on the front-end

Reply:
So the goal is if `$this->_viewFields` does not contain `plz` (because the user does not want to display it), but `$titleFields` contains it, then `plz` should _not_ be returned?

Reply:
Yes, that's right

Reply:
![image](https://user-images.githubusercontent.com/61687197/164135591-365881af-374c-415f-815f-e3910df660dd.png)

Reply:
![image](https://user-images.githubusercontent.com/61687197/164135680-8c06fcac-c12c-4fb9-b1f0-90751644152b.png)

Reply:
Doesn't `array_merge` put both together? So if `$this->_viewFields = ["plz", "gesamtflaeche"]` and `$titleFields = ["plz", "objekttitel"]`, then doesn't `array_values(array_unique(array_merge($this->_viewFields, $titleFields)))` return `["plz", "gesamtflaeche", "objekttitel"]`, but you want `["plz"]`?

It seems to work, so I don't want to be too nit-picky about this, but I'm just surprised that it works. :)

Reply:
yes, i will check it again

Reply:
@jmaas-onoffice I refactored the code, pls review it again, thanks